### PR TITLE
Ilham

### PR DESCRIPTION
**Pull request by ilham-saleh**

### CODE REVIEW

File: src/controllers/customer.js [hunk 1]
```diff
@@ -35,14 +35,48 @@ const createCustomer = async (req, res) => {
...
+      existingCustomer.contact = {
+        phone: contact.phone || existingCustomer.contact?.phone,
+        email: contact.email || existingCustomer.contact?.email,
+      };
+    }
+    const updatedCustomer = await updateCustomerDB(id, existingCustomer);
+    res.status(201).json({ customer: updatedCustomer });
+  } catch (error) {
+    console.log(error.message);
+  }
```
Comment (Uprising5034) [inline src/controllers/customer.js:76]: I would look at reviewing this try/catch block. Currently if any errors do happen the user gets no feedback from their request and will end up with a timeout. At the very least you should consider adding something along the lines of `res.status(500).json({error: "server error"})`. I would have expected the 404 res to also be part of the error handling rather than in the main `try` block.

File: src/controllers/customer.js [hunk 2]
```diff
@@ -35,14 +35,48 @@ const createCustomer = async (req, res) => {
...
+
+  try {
+    const existingCustomer = await findCustomerDB(id);
+    if (!existingCustomer) {
+      return res
+        .status(404)
+        .json({ error: "Customer with that id does not exist" });
+    }
+
+    existingCustomer.name = name || existingCustomer.name;
```
Comment (Uprising5034) [inline src/controllers/customer.js:64]: I like the or assignments, I always forget they exist!

File: src/controllers/customer.js [hunk 3]
```diff
@@ -35,14 +35,48 @@ const createCustomer = async (req, res) => {
...
+    return res.status(400).json({ error: "Missing field in request body" });
+  }
+
+  try {
+    const existingCustomer = await findCustomerDB(id);
+    if (!existingCustomer) {
+      return res
+        .status(404)
+        .json({ error: "Customer with that id does not exist" });
+    }
```
Comment (Uprising5034) [inline src/controllers/customer.js:62]: If the database is set up correctly, it should already be throwing errors if you give invalid ids in your `UPDATE` and `INSERT` queries. You therefore don't need to check with a `SELECT` query before proceeding with trying to update.

File: src/domains/screen.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const prisma = require("../utils/prisma.js");
+
+const createScreenDB = async (number) =>
+  await prisma.screen.create({
+    data: { number: number },
```
Comment (Uprising5034) [inline src/domains/screen.js:5]: This can be shortened to:
```js
data: { number },
```

File: src/domains/movie.js
Comment (Uprising5034) [inline src/domains/movie.js:1]: Looking good to me!

File: src/controllers/movie.js
Comment (Uprising5034) [inline src/controllers/movie.js:1]: Controllers looking solid, next step is adding error handling 😄

**Review by Uprising5034 — COMMENTED** [review]
Nice stuff! I'd look into adding tests, it will help you in figuring out your edge cases and how to handle them.

File: src/controllers/customer.js (hunk 3 above)
Comment (AllyDouillette) [reply src/controllers/customer.js:62]: To add to that – you might have to look up the proper Prisma internal error key, I think it's P2025 – https://www.prisma.io/docs/orm/reference/error-reference#p2025

**Review by AllyDouillette — COMMENTED** [review]
Good work, but make sure to enhance the controllers a bit so all cases (happy path, unhappy paths) get a clear response for clarity towards the user :)

File: src/domains/customer.js [hunk 5]
```diff
@@ -1,26 +1,58 @@
...
+    // This is like doing RETURNING in SQL
+    include: {
+      contact: true,
+    },
+  });
+
+const updateCustomerDB = async (id, data) =>
+  await prisma.customer.update({
+    where: {
+      id: Number(id),
```
Comment (Satokii) [inline src/domains/customer.js:28]: Good use of `Number` to make sure you're passing through the relevant data type

File: src/domains/movie.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+const updateMovieDB = async (id, title, runtimeMins) =>
+  await prisma.movie.update({
+    where: {
+      id: Number(id),
+    },
+    data: { title: title, runtimeMins: runtimeMins },
+    include: {
+      screenings: true,
+    },
+  });
```
Comment (Satokii) [inline src/domains/movie.js:43]: Code if succinct and easy to follow!

File: test/api/extensions/customer-ext.spec.js [hunk 7]
```diff
@@ -1,56 +1,56 @@
-const supertest = require("supertest")
-const app = require("../../../src/server.js")
-const { createCustomer } = require("../../helpers/createCustomer.js")
+const supertest = require("supertest");
```
Comment (Satokii) [inline test/api/extensions/customer-ext.spec.js:1]: Just need to write the extra extension tests to make sure you're handling all the queries correctly, looks all good otherwise.

File: src/routers/movie.js
Comment (irlydo) [inline src/routers/movie.js:1]: Nice consistent Router files that are clear and concise

File: src/domains/movie.js
Comment (irlydo) [inline src/domains/movie.js:1]: Again, clear and concise without any need for improvement that I can see

File: src/controllers/customer.js (hunk 2 above)
Comment (ilham-saleh) [reply src/controllers/customer.js:64]: haha, finally something I did good here

File: src/controllers/customer.js (hunk 3 above)
Comment (ilham-saleh) [reply src/controllers/customer.js:62]: hey, thank you both for the reference and feedback, I surely will update the code

**Comment by vherus** [comment]
Nice tests 👍
